blurring_as_a_service/cleaning_pipeline/source/sampling_pipeline.py

```diff
@@ -0,0 +1,458 @@
+        ValueError
+            If there is an error in processing or parsing the data.
+        Exception
+            For any other unexpected errors that might occur during the execution of the function.
+        """
+        
+        images_statistics = {}
+
+        try:
+            
+            # Connect to the database
+            database_parameters = json.loads(self.database_parameters_json)
+            db_username = database_parameters["db_username"]
+            db_name = database_parameters["db_name"]
+            db_hostname = database_parameters["db_hostname"]
+            
+            # Validate if database credentials are provided
+            if not db_username or not db_name or not db_hostname:
+                raise ValueError("Please provide database credentials.")
+            
+            # Create a DBConfigSQLAlchemy object
+            db_config = DBConfigSQLAlchemy(db_username, db_hostname, db_name)
+            
+            # Create the database connection
+            db_config.create_connection()
+            
+            conf_score_threshold = self.sampling_parameters["conf_score_threshold"]
+        
+            with db_config.managed_session() as session:
+                for upload_date, image_names in grouped_images_by_date.items():
+                    logger.info(f'Upload Date: {upload_date} \n')
+                    upload_date = datetime.strptime(upload_date, "%Y-%m-%d_%H_%M_%S")
+                    logger.info(f'Formatted Upload Date: {upload_date} \n')
+                    
+                    for image_name in image_names:
+                        query = session.query(DetectionInformation).filter(
+                            DetectionInformation.image_customer_name == self.customer_name,
+                            DetectionInformation.image_upload_date == upload_date,
+                            DetectionInformation.image_filename == image_name,
+                            DetectionInformation.conf_score > conf_score_threshold
+                        )
+                        results = query.all()
+                        count = len(results)
+
+                        # Populating images_statistics with detailed information
+                        if results:
+                            extracted_data = [result.__dict__ for result in results]
+                            for data in extracted_data:
+                                data['count'] = count
+                                
+                            if upload_date in images_statistics:
+                                images_statistics[upload_date].extend(extracted_data)
+                            else:
+                                images_statistics[upload_date] = extracted_data
+                                
+                # Flatten the dictionary to a list of dictionaries
+                flat_list = [item for sublist in images_statistics.values() for item in sublist]
```

Comment:
This is an example of something that needs a comment and therefore should be refactored into a function for clarity. Possibly in a utils file where if someone needs to redo this in the future can just call the same function.